packages/contracts-periphery/contracts/universal/op-nft/OptimistInviter.sol

```diff
@@ -0,0 +1,228 @@
+     *                        signature is that the issuer may be using a ERC-1271 compatible
+     *                        contract wallet, where the recovered address is not the same as the
+     *                        issuer, or the signature is not an ECDSA signature at all.
+     * @custom:field nonce    Pseudorandom nonce to prevent replay attacks.
+     */
+    struct ClaimableInvite {
+        address issuer;
+        bytes32 nonce;
+    }
+
+    /**
+     * @notice Maps from hashes to whether or not they have been committed.
+     */
+    mapping(bytes32 => bool) public commitments;
+
+    /**
+     * @notice Maps from addresses to nonces to whether or not they have been used.
+     */
+    mapping(address => mapping(bytes32 => bool)) public usedNonces;
+
+    /**
+     * @custom:semver 1.0.0
+     *
+     * @param _inviteGranter      Address of the invite granter.
+     * @param _attestationStation Address of the AttestationStation contract.
+     */
+    constructor(address _inviteGranter, AttestationStation _attestationStation) Semver(1, 0, 0) {
+        INVITE_GRANTER = _inviteGranter;
+        ATTESTATION_STATION = _attestationStation;
+    }
+
+    /**
+     * @notice Initializes the OptimistInviter contract, setting the EIP712 context.
+     *
+     * @param _name Contract name
+     */
+    function initialize(string memory _name) public reinitializer(VERSION) {
+        __EIP712_init(_name, version());
+    }
+
+    /**
+     * @notice Allows invite granter to set the number of invites an address has.
+     *
+     * @param _accounts    An array of accounts to update the invite counts of.
+     * @param _inviteCount Number of invites to set to.
+     */
+    function setInviteCounts(address[] calldata _accounts, uint256 _inviteCount) public {
+        // Only invite granter can grant invites
+        require(
+            msg.sender == INVITE_GRANTER,
+            "OptimistInviter: only invite granter can grant invites"
+        );
+
+        uint256 length = _accounts.length;
+
+        for (uint256 i; i < length; ) {
+            // The granted invites are stored as an attestation from this contract on the
+            // AttestationStation contract. Number of invites is stored as a encoded uint256 in the
+            // data field of the attetation.
+            ATTESTATION_STATION.attest(
+                _accounts[i],
+                bytes32("optimist.can-invite"),
+                abi.encode(_inviteCount)
+            );
+
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /**
+     * @notice Allows anyone to commit a received signature along with the address to claim to.
+     *         This is necessary to prevent front-running when the invitee is claiming the invite.
+     *
+     * @param _commitment A hash of the claimer and signature concatenated.
+                          keccak256(abi.encode(_claimer, _signature))
```

Comment:
```suggestion
    *                     keccak256(abi.encode(_claimer, _signature))
```

Reply:
Surprised we have no validation of natspec formatting.

Reply:
Will fix!